app/cities/views.py

```diff
@@ -1,25 +1,25 @@
-from rest_framework import permissions, status
-from rest_framework import viewsets
-from rest_framework.exceptions import ParseError
-
-from cities.models import City
-from cities.serializers import CitySerializer
-from app.enums import Region
-
-# Create your views here.
-
-
-class CityViewSet(viewsets.ReadOnlyModelViewSet):
-    permission_classes = (permissions.AllowAny,)
-    queryset = City.objects.all()
-    serializer_class = CitySerializer
-
-    def get_queryset(self):
-        queryset = self.queryset
-        region = self.request.query_params.get('region', None)
-        if region is not None:
-            if region not in Region.values:
-                raise ParseError(detail='Invalid region name. Only permitted names are: [nord, midt, vest, sør, øst]', code=status.HTTP_400_BAD_REQUEST)
-            else:
-                queryset = queryset.filter(region=region)
+from rest_framework import permissions, status
+from rest_framework import viewsets
+from rest_framework.exceptions import ParseError
+
+from cities.models import City
+from cities.serializers import CitySerializer
+from app.enums import Region
+
+# Create your views here.
+
+
+class CityViewSet(viewsets.ReadOnlyModelViewSet):
+    permission_classes = (permissions.AllowAny,)
+    queryset = City.objects.all()
+    serializer_class = CitySerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        region = self.request.query_params.get('region', None)
+        if region is not None:
+            if region not in Region.values:
+                raise ParseError(detail='Invalid region name. Only permitted names are: [nord, midt, vest, sør, øst]', code=status.HTTP_400_BAD_REQUEST)

```

Comment:
I find this to be somewhat misleading as I interpret ParseError to be something related to the actual parsing of the data, like an error between JSON and the Python code. This does not match my expectation here, as I would expect a ValueError. Is there any special reason that you use ParseError?

Reply:
Not sure why ParseError was used, but would NotFound be acceptable here? This would also fit nicely with the fact that it now returns a 404 and not 400 as mentioned in the other discussion.

Reply:
Absolutely! 👍